plugin/storage/grpc/shared/grpc_client_test.go

```diff
@@ -302,3 +317,138 @@ func TestGRPCClientGetDependencies(t *testing.T) {
 		assert.Equal(t, deps, s)
 	})
 }
+
+func TestGrpcClientWriteArchiveSpan(t *testing.T) {
+	withGRPCClient(func(r *grpcClientTest) {
+		r.archiveWriter.On("WriteArchiveSpan", mock.Anything, &storage_v1.WriteSpanRequest{
+			Span: &mockTraceSpans[0],
+		}).Return(&storage_v1.WriteSpanResponse{}, nil)
+
+		err := r.client.WriteArchiveSpan(&mockTraceSpans[0])
```

Comment:
I think this is a misleading test. Instead of calling WriteArchiveSpan, we should be calling `client.ArchiveWriter().WriteSpan`.